### ISSUE TITLE
require-readonly-react-props treats read only as sealed for classes

### ISSUE THREAD

**Issue by lukeapage** [gajus/eslint-plugin-flowtype#421]
The following does not give an error but it should:
```
type Props = $ReadOnly<{ 
    onX: (y: types.Y) => void,
    onY: (y: types.Y) => void,
    onZ: (y: types.Y) => void,
    // + all the props from abc 
}>;

  class A extends React.PureComponent<Props> {
```

**Comment by lukeapage** [comment]
@gajus @kangax Was looking at implementing this and realised it is an enhancement - its not 100% clear but the rule essentially just doesn't verify the type is exact.
Given flow will be changing the default, I guess this should be either a new rule or a new option?
If its a new option, is the rule name confusing?
So, I'm guessing it should be a new rule, and take some of the existing code into helper files?

I might just see if our codebase can just use the require-exact-type everywhere.

**Comment by kangax** [comment]
Ah yes, we have require-exact-type enabled in our codebase so I didn't realise linter needs to check for that too. Should be fairly easy to add support for checking exactness unless the type is defined elsewhere (i.e. `$ReadOnly<ExactProps>` vs. `$ReadOnly<{| ... |}>`)

**Comment by gajus** [comment]
I am not clear about what is the issue here.

`Props` is `$ReadOnly` and that's what the rule enforces. What is the reason exactness is needed here?
